File: packages/ui-compose/ui/src/main/kotlin/com/kinetixui/ui/Quote.kt
package com.kinetixui.ui

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.res.dimensionResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.sp

/**
 * KinetixQuote — mirrors `packages/ui/src/components/quote.tsx`: a
 * blockquote with an optional attributed author (name, title, avatar
 * slot). Curly quotes are added around the text here the same way the
 * React source adds them around `children` — not the caller's job.
 */
@Composable
fun KinetixQuote(
    text: String,
    modifier: Modifier = Modifier,
    author: String? = null,
    authorTitle: String? = null,
    avatar: (@Composable () -> Unit)? = null,
) {
    val colors = KinetixColorScheme.current
    Column(modifier = modifier, verticalArrangement = Arrangement.spacedBy(dimensionResource(R.dimen.spacing_4))) {
        Text(
            text = "“$text”",
            color = colors.foreground,
            fontWeight = FontWeight.Medium,
            fontSize = dimensionResource(R.dimen.font_size_title_md).value.sp,
            lineHeight = dimensionResource(R.dimen.line_height_title_md).value.sp,
        )
        if (author != null || authorTitle != null) {
            Row(
                horizontalArrangement = Arrangement.spacedBy(dimensionResource(R.dimen.spacing_3)),
                verticalAlignment = Alignment.CenterVertically,
            ) {
                avatar?.invoke()
                Column {
                    if (author != null) {
                        Text(
                            text = author,
                            color = colors.foreground,
                            fontWeight = FontWeight.Medium,
                            fontSize = dimensionResource(R.dimen.font_size_body_md).value.sp,
                            lineHeight = dimensionResource(R.dimen.line_height_body_md).value.sp,
                        )
                    }
                    if (authorTitle != null) {
                        Text(
                            text = authorTitle,
                            color = colors.mutedForeground,
                            fontSize = dimensionResource(R.dimen.font_size_body_sm).value.sp,
                            lineHeight = dimensionResource(R.dimen.line_height_body_sm).value.sp,
                        )
                    }
                }
            }
        }
    }
}

File: packages/ui-swiftui/Sources/KinetixUI/Quote.swift
//
// Quote.swift — KinetixQuote.
//
// Mirrors packages/ui/src/components/quote.tsx: a blockquote with an
// optional attributed author (name, title, avatar slot). Curly quotes are
// added around the text here, the same as the React source does around
// `children` — not the caller's job.
//

import SwiftUI

public struct KinetixQuote<Avatar: View>: View {
    @Environment(\.kinetixColors) private var colors

    private let text: String
    private let author: String?
    private let authorTitle: String?
    private let avatar: Avatar

    public init(
        _ text: String,
        author: String? = nil,
        authorTitle: String? = nil,
        @ViewBuilder avatar: () -> Avatar
    ) {
        self.text = text
        self.author = author
        self.authorTitle = authorTitle
        self.avatar = avatar()
    }

    public var body: some View {
        VStack(alignment: .leading, spacing: 16) {
            Text("\u{201C}\(text)\u{201D}")
                .font(.system(size: 18, weight: .medium)) // title-md
                .foregroundStyle(colors.foreground)
                .fixedSize(horizontal: false, vertical: true)
            if author != nil || authorTitle != nil {
                HStack(spacing: 12) {
                    avatar
                    VStack(alignment: .leading, spacing: 2) {
                        if let author {
                            Text(author)
                                .font(.system(size: 14, weight: .medium))
                                .foregroundStyle(colors.foreground)
                        }
                        if let authorTitle {
                            Text(authorTitle)
                                .font(.system(size: 13))
                                .foregroundStyle(colors.mutedForeground)
                        }
                    }
                }
            }
        }
    }
}

public extension KinetixQuote where Avatar == EmptyView {
    /// No avatar slot.
    init(_ text: String, author: String? = nil, authorTitle: String? = nil) {
        self.init(text, author: author, authorTitle: authorTitle) { EmptyView() }
    }
}
